lib/OpenCloud/LoadBalancer/Service.php

```diff
@@ -51,11 +52,20 @@ public function loadBalancer($id = null)
      */
     public function loadBalancerList($detail = true, array $filter = array())
     {
+        $options = $this->makeResourceIteratorOptions($this->resolveResourceClass('LoadBalancer'));
+
+        if (isset($filter['limit'])) {
+            $options['limit.page'] = $filter['limit'];
+            unset($filter['limit']);
+        }
+
         $url = $this->getUrl();
         $url->addPath(Resource\LoadBalancer::resourceName());
         $url->setQuery($filter);
 
-        return $this->resourceList('LoadBalancer', $url);
+        $options += array('baseUrl' => $url, 'key.marker' => 'id');
```

Comment:
The base URL that is [passed in as an option](https://github.com/jamiehannaford/php-opencloud/blob/lb-paginator-fix/lib/OpenCloud/LoadBalancer/Service.php#L66) to the iterator is subsequently [used to construct next URLs](https://github.com/jamiehannaford/php-opencloud/blob/lb-paginator-fix/lib/OpenCloud/Common/Collection/PaginatedIterator.php#L272). If the query value is set on the initial URL, then it should still be set when the iterator constructs the next one.